### PR TITLE
Check code formatting with autopep8, fix formatting for codebase

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**

### CODE REVIEW

**Comment by BenjaminRodenberg** [comment]
@IshaanDesai For review: Please take special care about b6e013a, since I touched the tests here. Hopefully did not break anything.

**Review by BenjaminRodenberg — COMMENTED** [review]
Ready for review. Already had a very close look at `test/test_bindings_module.py`. I'm quite hopeful that I did not break the tests.

File: solverdummy/solverdummy.py [hunk 1]
```diff
@@ -5,7 +5,8 @@
 import precice
 
 parser = argparse.ArgumentParser()
-parser.add_argument("configurationFileName", help="Name of the xml config file.", type=str)
+parser.add_argument("configurationFileName",
+                    help="Name of the xml config file.", type=str)
 parser.add_argument("participantName", help="Name of the solver.", type=str)
 parser.add_argument("meshName", help="Name of the mesh.", type=str)
```
Comment (IshaanDesai) [inline solverdummy/solverdummy.py:12]:
```suggestion
args = None
```

File: solverdummy/solverdummy.py [hunk 2]
```diff
@@ -33,7 +34,8 @@
 solver_process_index = 0
 solver_process_size = 1
 
-interface = precice.Interface(participant_name, configuration_file_name, solver_process_index, solver_process_size)
+interface = precice.Interface(participant_name, configuration_file_name,
+                              solver_process_index, solver_process_size)
 
 mesh_id = interface.get_mesh_id(mesh_name)
 dimensions = interface.get_dimensions()
```
Comment (IshaanDesai) [inline solverdummy/solverdummy.py:41]:
```suggestion
read_data_name, write_data_name = None, None
```

File: solverdummy/solverdummy.py [hunk 3]
```diff
@@ -33,7 +34,8 @@
 solver_process_index = 0
 solver_process_size = 1
 
-interface = precice.Interface(participant_name, configuration_file_name, solver_process_index, solver_process_size)
+interface = precice.Interface(participant_name, configuration_file_name,
+                              solver_process_index, solver_process_size)
```
Comment (IshaanDesai) [inline solverdummy/solverdummy.py:39]:
```suggestion
read_data_name, write_data_name = None, None
```
This line can go even further below but I was not able to add a suggestion where there was no code change in the review.

**Review by IshaanDesai — APPROVED** [review]
In `setup.py` function names like `my_build_ext` and `my_install` prompt a suggestion *Class names should use CamelCase convention*. I am not sure if this is relevant and important.

**Comment by BenjaminRodenberg** [comment]
> 
> 
> In `setup.py` function names like `my_build_ext` and `my_install` prompt a suggestion _Class names should use CamelCase convention_. I am not sure if this is relevant and important.

Technically yes. However, I think this is a corner case, since also setuptools diverges from this convention:

https://github.com/pypa/setuptools/blob/8222d6f7b992d3b184434acb31cd66b0f2e41401/setuptools/command/build_ext.py#L75

So let's keep it as it is. Thanks for the review! I will merge now.

**Pull request merged**